codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowControllerTest.java

```diff
@@ -1109,26 +1107,21 @@ public Void answer(InvocationOnMock invocationOnMock) {
     public void flowControlledWriteCompleteThrowsAnException() throws Exception {
         final Http2RemoteFlowController.FlowControlled flowControlled =
                 Mockito.mock(Http2RemoteFlowController.FlowControlled.class);
-        when(flowControlled.size()).thenReturn(100);
-        when(flowControlled.write(anyInt())).thenAnswer(new Answer<Boolean>() {
-            private int invocationCount;
+        final AtomicInteger size = new AtomicInteger(150);
+        doAnswer(new Answer<Integer>() {
             @Override
-            public Boolean answer(InvocationOnMock invocationOnMock) throws Throwable {
-                switch(invocationCount) {
-                    case 0:
-                        when(flowControlled.size()).thenReturn(50);
-                        invocationCount = 1;
-                        return true;
-                    case 1:
-                        when(flowControlled.size()).thenReturn(20);
-                        invocationCount = 2;
-                        return true;
-                    default:
-                        when(flowControlled.size()).thenReturn(0);
-                        return false;
-                }
+            public Integer answer(InvocationOnMock invocationOnMock) throws Throwable {
+                return size.get();
+            }
+        }).when(flowControlled).size();
+        doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
+                size.addAndGet(-50);
```

Comment:
Probably not, looking at it more closely. I probably could have just converted it to Answer<Void>